File: Bookstore.Core/Models/Orders/OrderModel.cs
using Bookstore.Core.Models.Payments;
using Bookstore.Core.Models.Users;
using System;
using System.Collections.Generic;
using static Bookstore.Shared.Enums.Enums;

namespace Bookstore.Core.Models.Orders
{
    public class OrderModel : Model
    {
        public Guid UserId { get; set; }

        public UserModel User { get; set; }

        public Guid PaymentId { get; set; }

        public PaymentModel Payment { get; set; }

        public string Description { get; set; }

        public OrderStatusType Status { get; set; }

        public double TotalCost { get; set; }

        public List<OrderItemModel> Items;

        public OrderModel() => 
            Items = new List<OrderItemModel>();
    }
}

File: Bookstore.Web/ClientApp/src/app/models/order/order-model.ts
import { OrderStatus } from "src/app/enums/order-status";
import { Payment } from "../payment/payment-model";
import { User } from "../user/user.model";
import { OrderItem } from "./order-item-model";


export class Order
{
  id: string
  userId: string
  user: User
  date: Date
  paymentId: string
  payment: Payment
  status: OrderStatus
  totalCost: number
  items: OrderItem[]
}
